### PR TITLE
Initial checkin of rna-quantification (salmon) workflow.

### LINKED ISSUE #2: Add Salmon workflow
For AMP PD RNA data, a salmon workflow was run in Terra.
Add to this repository:

- README describing the setup of the index file
- WDL
- inputs.json

### PR DESCRIPTION
**Pull request by wnojopra**
This PR includes the rna-quantification workflow which executes salmon on a pair of fastq files.

Tested in our test workspace.

Fixes #2 .

### CODE REVIEW

File: rna/rna-quantification/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# RNASummarization with subread featureCounts
```
Comment (mbookman) [inline rna/rna-quantification/README.md]: # RNAQuantification with salmon quant

File: rna/rna-quantification/rna-quantification.inputs.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "##_COMMENT1": "Set your per-sample values: name and fastq files",
+  "RNAQuantification.sample_name": "${this.sample_id}",
+  "RNAQuantification.fastq_1": "${this.fastq2}",
+  "RNAQuantification.fastq_2": "${this.fastq1}",
+
+  "##_COMMENT2": "Path to the reference GTF file and salmon index file",
+  "RNAQuantification.gene_map": "gs://amp-pd-transcriptomics/inputs/reference/gencode/gencode.v29.primary_assembly.annotation.gtf.gz",
+  "RNAQuantification.salmon_index": "gs://amp-pd-data-rna-workflows/inputs/reference/salmon/Homo_sapiens.gencode.v29.all.salmon0.11.3_gencodeOption.index.tar.gz",
```
Comment (wnojopra) [inline rna/rna-quantification/rna-quantification.inputs.json]: Matt, do I need to move this to gs://amp-pd-transcriptomics?

File: rna/rna-quantification/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+- Per-sample
+  - Transcript Quantification files
+  - Gene Quantification files
+  - Command info
+  - Parameter files
+  - Log files
+  - Auxilliary files
+
+## Notes
+A sample inputs.json file is included here with values derived from running workflows for AMP PD on [Terra.](https://app.terra.bio/)
```
Comment (mbookman) [inline rna/rna-quantification/README.md]: The period should be outside of the hyperlink.

File: rna/rna-quantification/rna-quantification.inputs.json (hunk 2 above)
Comment (mbookman) [inline rna/rna-quantification/rna-quantification.inputs.json]: gs://amp-pd-transcriptomics/inputs/reference/salmon/Homo_sapiens.gencode.v29.all.salmon0.11.3_gencodeOption.index.tar.gz

File: rna/rna-quantification/rna-quantification.inputs.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+  "##_COMMENT3": "Path to a Docker image containing salmon",
+  "RNAQuantification.salmon_docker": "combinelab/salmon:0.11.3",
+  
+  "RNAQuantification.runtime_zones": "us-central1-c us-central1-b",
```
Comment (mbookman) [inline rna/rna-quantification/rna-quantification.inputs.json]: Let's be consistent with the zones we include as examples.
Let's standardize on `us-central1-a us-central1-b` in the inputs.json and the README.md.

File: rna/rna-quantification/rna-quantification.wdl [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    set -o nounset
+    set -o pipefail
+    
+    # Extract salmon index files to its own directory.
+    mkdir salmon_index
+    tar xfz "${salmon_index}" --directory salmon_index
+
+    # Unzip the gene map.
+    gunzip "${gene_map}"
+    UNZIPPED_GENE_MAP="$(echo ${gene_map} | sed -e "s/.gz$//")"
```
Comment (mbookman) [inline rna/rna-quantification/rna-quantification.wdl]: Change the sed expression to be enclosed by single quotes.
Single quotes suppresses $ expansion.

```
UNZIPPED_GENE_MAP="$(echo ${gene_map} | sed -e 's/.gz$//')"
```

**Review by mbookman — CHANGES_REQUESTED** [review]

**Comment by wnojopra** [comment]
Thanks, all fixes committed.

File: rna/rna-quantification/README.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
+# RNAQuantification with salmon quant
+
+This workflow runs the salmon command on a pair of FASTQ files
```
Comment (mbookman) [inline rna/rna-quantification/README.md]: This workflow runs the `salmon quant` command on a pair of FASTQ files

File: rna/rna-quantification/rna-quantification.inputs.json [hunk 7]
```diff
@@ -0,0 +1,20 @@
+{
+  "##_COMMENT1": "Set your per-sample values: name and fastq files",
+  "RNAQuantification.sample_name": "${this.sample_id}",
+  "RNAQuantification.fastq_1": "${this.fastq2}",
+  "RNAQuantification.fastq_2": "${this.fastq1}",
```
Comment (mbookman) [inline rna/rna-quantification/rna-quantification.inputs.json]:
```
  "RNAQuantification.sample_name": "__INSERT_SAMPLE_NAME__",
  "RNAQuantification.fastq_1": [ "__INSERT_GCS_PATHS_TO_FASTQ1__" ],
  "RNAQuantification.fastq_2": [ "__INSERT_GCS_PATHS_TO_FASTQ2__" ],
```

**Review by mbookman — CHANGES_REQUESTED** [review]

File: rna/rna-quantification/README.md (hunk 6 above)
Comment (mbookman) [reply rna/rna-quantification/README.md]: Actually - should be "on a list of paired FASTQ files"

**Review by mbookman — APPROVED** [review]
LGTM

File: rna/rna-quantification/README.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  - Gene Quantification files
+  - Command info
+  - Parameter files
+  - Log files
+  - Auxilliary files
+
+## Notes
+A sample inputs.json file is included here with values derived from running workflows for AMP PD on [Terra](https://app.terra.bio/).
+
+The gene map (GTF) file used was [GENCODE](https://www.gencodegenes.org/) v29.
```
Comment (mbookman) [inline rna/rna-quantification/README.md:36]: Let's add:

The Salmon index is packaged as a gzipped TAR file as WDL draft-2 does not support the WDL Directory type.
The Salmon index was created with the following steps:


1- Download the GENCODE v29 FASTA file:

```
ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_29/gencode.v29.transcripts.fa.gz
```

2- Run "salmon index" as:

```
salmon index \
  -t gencode.v29.transcripts.fa.gz \
  --gencode \
  -i Homo_sapiens.gencode.v29.all.salmon0.11.3_gencodeOption.index \
  --type quasi \
  -k 31
```

3- Build a gzipped TAR file for the Salmon workflow to pull just a single reference file:

```
tar cvfz Homo_sapiens.gencode.v29.all.salmon0.11.3_gencodeOption.index.tar.gz Homo_sapiens.gencode.v29.all.salmon0.11.3_gencodeOption.index/
```

**Review by mbookman — APPROVED** [review]
LGTM

**Pull request merged**